app/routes/lending/useFetchedLendableObject.ts

```diff
@@ -0,0 +1,38 @@
+import { usePreparedEffect } from '@webkom/react-prepare';
+import { useState } from 'react';
+import { fetchLendableObjectById } from 'app/actions/LendableObjectActions';
+import { selectLendableObjectById } from 'app/reducers/lendableObjects';
+import { useAppDispatch, useAppSelector } from 'app/store/hooks';
+import type { EntityId } from '@reduxjs/toolkit';
+import type {
+  DetailLendableObject,
+  UnknownLendableObject,
+} from 'app/store/models/LendableObject';
+
+export const useFetchedLendableObject = (lendableObjectId: EntityId) => {
+  const dispatch = useAppDispatch();
+  const [fetching, setFetching] = useState(true);
+
+  usePreparedEffect(
+    'fetchLendableObjectById',
+    () =>
+      dispatch(fetchLendableObjectById(lendableObjectId)).then(() =>
+        setFetching(false),
+      ),
+    [],
```

Comment:
```suggestion
    [lendableObjectId],
```
Shouldn't the id be in the dependency array?

Reply:
Nice catch! We should get the `useEffect` dependency linter to lint these too